src/api/spec/components/diff_subject_component_spec.rb

```diff
@@ -12,4 +12,39 @@
       expect(rendered_content).to have_text(preview_name.capitalize)
     end
   end
+
+  context 'when the state is added' do
+    it 'renders the new filename' do
+      render_inline(described_class.new(state: 'added', file_info: { 'new' => { 'name' => 'new_file.txt' }, 'old' => { 'name' => '' } }))
+      expect(rendered_content).to have_text('new_file.txt')
+    end
+  end
+
+  context 'when the state is deleted' do
+    it 'renders the new filename' do
+      render_inline(described_class.new(state: 'deleted', file_info: { 'new' => { 'name' => '' }, 'old' => { 'name' => 'old_file.txt' } }))
+      expect(rendered_content).to have_text('old_file.txt')
+    end
+  end
+
+  context 'when the state is changed' do
+    it 'renders the new filename' do
+      render_inline(described_class.new(state: 'changed', file_info: { 'new' => { 'name' => 'new_file.txt' }, 'old' => { 'name' => 'new_file.txt' } }))
+      expect(rendered_content).to have_text('new_file.txt')
+    end
+  end
+
+  context 'when the state is renamed' do
+    it 'renders the new filename' do
+      render_inline(described_class.new(state: 'renamed', file_info: { 'new' => { 'name' => 'new_file.txt' }, 'old' => { 'name' => 'old_file.txt' } }))
+      expect(rendered_content).to have_text('old_file.txt')
```

Comment:
```suggestion
      expect(rendered_content).to have_text('old_file.txt')
      expect(rendered_content).to have_text('new_file.txt')
```
For the `renamed` case, shouldn't make sense to add also a check for having the new_file.txt text?